### ISSUE TITLE
Add HTTPS Pinning

### ISSUE THREAD

**Issue by raigu** [SK-EID/smart-id-php-client#19]
Smart ID documentation covers[ HTTPS pinning](https://github.com/SK-EID/smart-id-documentation/blob/master/README.md#35-api-endpoint-authentication). Unfortunately it is a little bit tricky and would be nice to have in this package.

I see that @andrevka has created a pullrequest #18 to address the issue.  It is create work but has one drawback. It breaks the backward compatibility. Currently package supports PHP >= 5.6 but with proposed pull request the new minimum requirement is >= 7.0.7

I had an idea how to make it work even with PHP 5.6 and created a branch for this development: https://github.com/raigu/smart-id-php-client/commits/pinning

I have a question for maintainer. Is it important to keep backward compatibility? If yes, should I continue with the referred branch? 

My proposed package is not finished. I made this issue to share my thought with @andrevka and maintainer and see if we have mutual interests and can cooperate.  What I have not done yet:
- documentation with samples
- simplified keys (currently uses PEM but I see @andrevka has used hases, it is shorter)
- think through how to avoid conditional statements in end application when switching between production and demo environment
- if detected PHP > 7.0.7 environment then use curl's native pinning functionality

Will wait feedback from maintainer or @andrevka. If there is interest I/we can continue.

**Comment by raigu** [comment]
I have to admit that the backward compatibility after #18 actually remains. Just pinning feature is available only when PHP >= 7.0.7.  Therefore my concern is probably not much relevant anymore.
